File: android/PlayKit/app/src/main/java/io/streamroot/dna/samples/playkit/PlayKitInteractor.java
package io.streamroot.dna.samples.playkit;

import android.os.Looper;
import com.kaltura.android.exoplayer2.LoadControl;
import com.kaltura.playkit.Player;
import io.streamroot.dna.core.PlayerInteractor;
import io.streamroot.dna.core.TimeRange;

import java.util.Collections;
import java.util.List;

final class PlayKitInteractor implements PlayerInteractor {
    private Player mPlayer;
    private DynamicBufferLoadControl mLoadControlWrapper;
    PlayKitInteractor(Player player, LoadControl loadControl) {
        mPlayer = player;
        mLoadControlWrapper = new DynamicBufferLoadControl(loadControl);
    }

    @Override
    public double bufferTarget() {
        return mLoadControlWrapper.bufferTarget();
    }

    @Override
    public void setBufferTarget(double target) {
        mLoadControlWrapper.setBufferTarget(target);
    }

    @Override
    public List<TimeRange> loadedTimeRanges() {
        long currentPosition = mPlayer.getCurrentPosition();
        long bufferedPosition = mPlayer.getBufferedPosition();
        return Collections.singletonList(new TimeRange(currentPosition, bufferedPosition));
    }

    @Override
    public Looper looper() {
        return Looper.getMainLooper();
    }

    @Override
    public long playbackTime() {
        return mPlayer.getCurrentPosition();
    }
}

File: android/PlayKit/app/src/main/java/io/streamroot/dna/playkit/PlaykitInteractor.kt
package io.streamroot.dna.playkit

import android.os.Looper
import com.kaltura.playkit.Player
import io.streamroot.dna.core.PlayerInteractor
import io.streamroot.dna.core.TimeRange
import java.util.*

class PlayKitInteractor(
        player : Player
) : PlayerInteractor {
    private val mPlayer = player

    override fun looper(): Looper? = Looper.getMainLooper()

    override fun playbackTime() : Long {
        return mPlayer.currentPosition
    }

    override fun loadedTimeRanges(): List<TimeRange> {
        val currentPosition = mPlayer.currentPosition
        val bufferedPosition = mPlayer.bufferedPosition
        return Collections.singletonList(TimeRange(currentPosition, bufferedPosition))
    }

    override fun bufferTarget(): Double {
        return mPlayer.bufferTarget ?: 0.0
    }

    override fun setBufferTarget(target: Double) {
        mPlayer.bufferTarget = target
    }
}